controller/controller_server/csi_controller_server.py

```diff
@@ -343,9 +343,53 @@ def CreateSnapshot(self, request, context):
             return csi_pb2.CreateSnapshotResponse()
 
     def DeleteSnapshot(self, request, context):
-        # TODO
-        logger.info("Delete snapshot")
-        return csi_pb2.DeleteSnapshotResponse()
+        set_current_thread_name(request.snapshot_id)
+        logger.info("DeleteSnapshot")
+        secrets = request.secrets
+
+        try:
+            utils.validate_delete_snapshot_request(request)
+
+            user, password, array_addresses = utils.get_array_connection_info_from_secret(secrets)
+
+            try:
+                array_type, snapshot_id = utils.get_volume_id_info(request.snapshot_id)
```

Comment:
add to utils function get_snapshot_id_info(...) and use it

Reply:
should I make a generic function to be used by both functions?
and if so, should it:
1. take the object type as a second parameter (like "_get_snapshot_name")? - this will require a refactor to make object type a const + map it to exceptions, etc...
or 
2. copy paste the code (like the first half of "get_snapshot")?

Reply:
done